crates/sparrow-runtime/src/execute/output.rs

```diff
@@ -40,13 +37,49 @@ pub enum Error {
 
 impl error_stack::Context for Error {}
 
+/// The output destination.
+///
+/// TODO: Replace the protobuf destinations with pure Rust structs.
+#[derive(Debug)]
+pub enum Destination {
+    ObjectStore(ObjectStoreDestination),
+    #[cfg(feature = "pulsar")]
+    Pulsar(PulsarDestination),
+    Channel(tokio::sync::mpsc::Sender<RecordBatch>),
+}
```

Comment:
This is nice. Can pretend this is the start of our proto -> rust type layer :))

Reply:
Yeah. Although, I think it would be better if this was extensible. Eg.

```
trait Destination {
  fn write(stream: BoxStream<'static, RecordBatch>) -> ...
  // or
  fn write(batch: RecordBatch) -> ...
}

struct Pulsar { ... }
impl Destination for Pulsar { ... }
```

Ideally, that would let us (a) add destinations without extending the enum, allowing user-defined destinations and (b) let us put the plusar deps in a later crate in the compilation tree, tec.